test/plugins/login.test.js

```diff
@@ -148,102 +148,136 @@ describe('login plugin test', () => {
+            ));
+
+            describe('with whitelist', () => {
+                beforeEach(() => {
+                    whitelistServer = new hapi.Server();
+                    whitelistServer.app.userFactory = userFactoryMock;
+
+                    whitelistServer.connection({
+                        port: 1234
+                    });
+
+                    return whitelistServer.register({
+                        register: plugin,
+                        options: {
+                            password,
+                            oauthClientId: 'oauth_client_id',
+                            oauthClientSecret: 'oauth_client_secret',
+                            jwtPrivateKey: '1234secretkeythatissupersecret5678',
+                            https: false,
+                            whitelist: ['batman']
+                        }
+                    });
+                });
+
+                afterEach(() => {
+                    whitelistServer = null;
                 });
-            });
 
-            it('returns forbidden for invalid user', (done) => {
-                server.inject({
-                    url: '/login',
-                    credentials: {
-                        profile: {
-                            username: 'dne'
+                it('returns forbidden for non-whitelisted user', () => (
+                    whitelistServer.inject({
+                        url: '/login',
+                        credentials: {
+                            profile: {
+                                username: 'dne'
+                            }
                         }
-                    }
-                }, (reply) => {
-                    assert.equal(reply.statusCode, 403, 'Login route should be available');
-                    assert.notOk(reply.result.token, 'Token not returned');
-                    done();
+                    }).then((reply) => {
+                        assert.equal(reply.statusCode, 403, 'Login route should be available');
+                        assert.notOk(reply.result.token, 'Token not returned');
+                    })
+                ));
+
+                it('returns 200 for whitelisted user', () => {
+                    userFactoryMock.get.resolves(null);
+
+                    return whitelistServer.inject(options).then((reply) => {
+                        assert.equal(reply.statusCode, 200, 'Login route should be available');
+                        assert.ok(reply.result.token, 'Token not returned');
+                        assert.calledWith(userFactoryMock.get, { username });
+                        assert.calledWith(userFactoryMock.create, {
+                            username,
+                            token,
+                            password
+                        });
+                    });
                 });
             });
         });
 
-        it('POST exists', (done) => {
+        it('POST exists', () => (
             server.inject({
```

Comment:
If i put `return server.inject` eslint will complain that it's redundant.  Instead I turned it into a short-hand return, see the `(` above this line.